Cabal/src/Distribution/Simple/GHC/Internal.hs

```diff
@@ -283,9 +283,24 @@ getGhcInfo verbosity _implInfo ghcProg = do
   case reads xs of
     [(i, ss)]
       | all isSpace ss ->
-          return i
+          return $ detectLLd i
     _ ->
       dieWithException verbosity CantParseGHCOutput
+  where
+    -- The `lld` linker cannot create merge objects so we want to add a new key
+    -- to this map specifying whether the linker can create merge objects.
+    detectLLd :: [(String, String)] -> [(String, String)]
```

Comment:
I am not familiar with this part of the code base but it looks like we can detect features of the linker when we "configure" it. E.g. `programProperties` in https://hackage.haskell.org/package/Cabal-3.10.1.0/docs/Distribution-Simple-Program-Types.html#t:ConfiguredProgram

I'll check if this helps us.